### PR TITLE
Hw 5.3

### PR DESCRIPTION
**Pull request by fruts**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
 /**
  * <p>Дано файл, потрібно прочитати його вміст і вибрати всі слова що починаються на `w`.
  * Результат повернути у вигляді відсортованого масиву (за замовчуванням). Всі слова повинні
  * бути в нижньому регістрі. У випадку якщо таких слів не знайдено повернути пустий масив.
  * Приклад: Width world Wide web
  * Результат: web wide width world</p>
  */
+
 public class FileWork {
+    public static final String BIG_W = "W";
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/FileWork.java:17]:
```suggestion
    public static final String CHARACTER = "W";
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
+    public static final String SMALL_W = "w";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        StringBuilder scannedWords = new StringBuilder();
+        try (Scanner scan = new Scanner(file)) {
+            while (scan.hasNext()) {
+                String oneString = scan.next();
+                if (oneString.startsWith(SMALL_W) || oneString.startsWith(BIG_W)) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/FileWork.java:26]: try to solve with a single `startsWith` call

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -1,14 +1,42 @@
...
+                    scannedWords.append(oneString
+                            .toLowerCase()
+                            .replaceAll("[^a-z]", ""))
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/FileWork.java:29]: do you really need this `replaceAll`?

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -1,14 +1,42 @@
...
+                            .append(" ");
+                }
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/FileWork.java:34]: don't ignore exceptions. see the checklist

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -1,14 +1,38 @@
...
+                    scannedWords.append(oneString.replaceAll("\\W", "")).append(" ");
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Неверно указано имя файла, или его не существует", e);
+        }
+        return scannedWords.length() > 0
+                ? Arrays.stream(scannedWords.toString().trim()
+                .split(" ")).sorted().toArray(String[]::new)
+                : new String[0];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/FileWork.java:35]:
```suggestion
        return scannedWords.toString().split(" ");
```
this should work, if not - change your code. sorry, didn't notice at first review

Comment (fruts) [reply src/main/java/core/basesyntax/FileWork.java]: your option does not work

Comment (Tarabarnik) [reply src/main/java/core/basesyntax/FileWork.java]: yes, but they won't be sorted. try to use Arrays.sort() with a result of that split method

**Pull request closed without merging**